components/cdc/src/service.rs

```diff
@@ -159,21 +159,25 @@ impl ChangeData for Service {
                 // The size of the response should not exceed CDC_MAX_RESP_SIZE.
                 // Split the events into multiple responses by CDC_MAX_RESP_SIZE here.
                 let events_len = events.len();
-                let mut event_vecs = vec![Vec::with_capacity(events_len)];
+                let mut resp_vecs = Vec::with_capacity(events_len);
+                resp_vecs.push(ChangeDataEvent::default());
                 let mut current_events_size = 0;
                 for (size, event) in events {
-                    if current_events_size + size >= CDC_MAX_RESP_SIZE {
-                        event_vecs.push(Vec::with_capacity(events_len));
+                    if current_events_size >= CDC_MAX_RESP_SIZE {
+                        resp_vecs.push(ChangeDataEvent::default());
                         current_events_size = 0;
                     }
                     current_events_size += size;
```

Comment:
How about just calling [protobuf::Message::compute_size](https://tikv.github.io/doc/protobuf/trait.Message.html#tymethod.compute_size) to get its size?